config/tf_modules/aws-eks/default_node_group.tf

```diff
@@ -41,9 +41,28 @@ resource "random_id" "key_suffix" {
     spot_instances     = var.spot_instances
     node_disk_size     = var.node_disk_size
     node_instance_type = var.node_instance_type
+    launch_template    = try(var.node_launch_template["user_data"], "")
   }
 }
 
+resource "aws_launch_template" "eks_node" {
+  count         = (length(var.node_launch_template) > 0) ? 1 : 0
+  instance_type = var.node_instance_type
+  name_prefix   = "opta-${var.layer_name}"
+  # https://github.com/hashicorp/terraform-provider-aws/issues/15118
+
+  block_device_mappings {
+    device_name = "/dev/sda1"
+
+    ebs {
+      volume_size = var.node_disk_size + 10
```

Comment:
Why +10?

Reply:
A realization that disk is _really_ cheap and that extra space will be needed for the system. Can revert if you don't like

Reply:
I think we should revert. If someone explicitly asks for 10G they should get exactly that.